server/service/src/invoice/outbound_shipment/update/generate.rs

```diff
@@ -77,7 +81,7 @@ fn should_update_batches(
         let new_invoice_status_index = new_invoice_status.index();
 
         new_invoice_status_index >= InvoiceRowStatus::Picked.index()
-            && invoice_status_index < new_invoice_status_index
+            && invoice_status_index < InvoiceRowStatus::Picked.index()
```

Comment:
This is a case where I find the verbose version easier to read - this is pretty clear what the behaviour is, and is fewer lines of code

```
fn should_update_batches_total_number_of_packs(
    invoice: &InvoiceRow,
    status: &Option<UpdateOutboundShipmentStatus>,
) -> bool {
    match (&invoice.status, status) {
        (InvoiceRowStatus::New, Some(UpdateOutboundShipmentStatus::Picked)) => true,
        (InvoiceRowStatus::New, Some(UpdateOutboundShipmentStatus::Shipped)) => true,
        (InvoiceRowStatus::Allocated, Some(UpdateOutboundShipmentStatus::Picked)) => true,
        (InvoiceRowStatus::Allocated, Some(UpdateOutboundShipmentStatus::Shipped)) => true,
        _ => false,
    }
}
```

Reply:
@clemens-msupply has a similar comment, I thought it was ok, but can see it looks better with match:

I took a little break from reviewing and made a branch to see how it look for all of the places where invoice status index is used (I think it's a little bit too much for main branch, will PR to develop when this is merged and when main is merged into develop) [i think it does look better](https://github.com/openmsupply/open-msupply/compare/358-Fix-bug-with-reduction-of-total-number-of-packs-in-outbound-shipment-status-change...Change-status-check-to-use-match)